app/controllers/rss_logs_controller.rb

```diff
@@ -12,29 +13,19 @@ class RssLogsController < ApplicationController
   ]
   before_action :disable_link_prefetching
 
-  # Default page.  Just displays latest happenings.  The actual action is
+  # Default page. Just displays latest happenings. The actual action is
   # buried way down toward the end of this file.
   # Displays matrix of selected RssLog's (based on current Query, if exists).
   def index
     # POST requests with param `type` potentially show an array of types
     # of objects. The array comes from the checkboxes in tabset
-    if params[:type].present?
-      types = Array(params[:type])
-      types = RssLog.all_types.intersection(types)
-      types = "all" if types.length == RssLog.all_types.length
-      types = "none" if types.empty?
-      types = types.map(&:to_s).join(" ") if types.is_a?(Array)
-      query = find_or_create_query(:RssLog, type: types)
-    # Previously saved query, incorporating type and other params
-    elsif params[:q].present?
-      query = find_query(:RssLog)
-      query ||= create_query(:RssLog, :all,
-                             type: @user ? @user.default_rss_type : "all")
-    # Fresh version of the index, no existing query
-    else
-      query = create_query(:RssLog, :all,
-                           type: @user ? @user.default_rss_type : "all")
-    end
+    query = if params[:type].present?
+              type_query
+            elsif params[:q].present?
+              saved_or_new_query
+            else
+              new_query
+            end
```

Comment:
It might be possible to use the dispatch table pattern of #1307, 
```ruby
    query = super
    show_selected_rss_logs(query, id: params[:id].to_s, always_index: true)
```
But
- Perhaps too much trouble for just 3 cases
- I thought it looked weird: (and was unsure that it would work):